File: backend/app/schemas/signal.py
from typing import Optional
from app.schemas.base import BaseSchema


class SignalReason(BaseSchema):
    factor: str
    impact: str  # bullish, bearish, neutral
    detail: str


class TradeSignal(BaseSchema):
    action: str  # BUY, SELL, HOLD, NEUTRAL
    direction: str  # bullish, bearish, neutral
    confidence: float
    reasons: list[SignalReason]
    generated_at: str


class QuoteSummary(BaseSchema):
    price: float
    change: float
    change_percent: float


class SignalResponse(BaseSchema):
    signal: TradeSignal
    quote: QuoteSummary

File: frontend/src/types/signal.ts
export type SignalAction = "BUY" | "SELL" | "HOLD" | "NEUTRAL";
export type SignalDirection = "bullish" | "bearish" | "neutral";

export interface SignalReason {
  factor: string;
  impact: "bullish" | "bearish" | "neutral";
  detail: string;
}

export interface TradeSignal {
  action: SignalAction;
  direction: SignalDirection;
  confidence: number;
  reasons: SignalReason[];
  generatedAt: string;
}

export interface SignalResponse {
  signal: TradeSignal;
  quote: {
    price: number;
    change: number;
    changePercent: number;
  };
}
